Documentation Improvements

Freddyminu opened the pull request:
- Enhance README
- Add comprehensive examples for TagoIO Custom Widget SDK
- Add additional coments on custom-widget.ts

Jira Card:
https://tago-io.atlassian.net/browse/LCD-121

Freddyminu committed:
feat:
- Enhance README
- Add comprehensive examples for TagoIO Custom Widget SDK
- Add additional coments on custom-widget.ts

Freddyminu commented:
@gustavo-aguiar spoke with me and explained that the script is currently injecting html code which isn't safe practice as the AI pr review also mentioned. 

Must change to follow the screenshot below: 

<img width="674" height="500" alt="image" src="https://github.com/user-attachments/assets/89fe24c3-e3bb-4ac2-a15f-b3bb0d98c79a" />

Freddyminu committed:
PR Example

Freddyminu committed:
PR Example

Freddyminu committed:
refactor: update examples to HTML format and enhance README

- Converted JavaScript example files to HTML format for better usability.
- Updated README to reflect the changes in example file formats and provide clearer instructions.
- Added new examples for reading data, entity data handling, resource access, and time intervals.
- Removed outdated JavaScript files and ensured all examples are self-contained HTML documents.

gustavo-aguiar commented:
@Freddyminu LGTM in general, but would follow some instruction from copilot about the innerHTML

Never insert user-controlled content using .innerHTML.
Use .textContent or .innerText when outputting data.

Example:
```
const safeOutput = document.createTextNode(userInput);
document.getElementById("output").appendChild(safeOutput);
```

Freddyminu committed:
refactor: simplify example widgets and improve UI consistency

- Updated example widget HTML files to enhance readability and maintainability.
- Streamlined styles and structure for better user experience across all examples.
- Improved error handling and messaging for clarity.
- Removed redundant code and optimized JavaScript functions for performance.

vitorfdl commented on README.md:
```diff
@@ -1,29 +1,369 @@
+
+The TagoIO Custom Widget SDK provides a comprehensive API through the global `window.TagoIO` object:
+
+#### Initialization Functions
+- **`TagoIO.ready(options)`** - Signal that your widget is ready to start
+- **`TagoIO.onStart(callback)`** - Handle widget startup and receive configuration
+- **`TagoIO.onError(callback)`** - Handle API errors and widget issues
```
You must explain those are asynchronous events. 

The `.ready` "Signal that your widget is ready to start" for example is not clear, would I use it for sending a signal or would I receive the signal? What does it matter?

vitorfdl commented on README.md:
```diff
@@ -1,29 +1,369 @@
+- **`TagoIO.onStart(callback)`** - Handle widget startup and receive configuration
+- **`TagoIO.onError(callback)`** - Handle API errors and widget issues
+
+#### Data Operations
+- **`TagoIO.sendData(data, callback)`** - Send data to TagoIO devices
+- **`TagoIO.editData(data, callback)`** - Edit existing device data
+- **`TagoIO.deleteData(data, callback)`** - Delete device data
+- **`TagoIO.editResourceData(data, callback)`** - Edit platform resources
```
You shall explain those are heavily dependant of the Custom Widget configuration at TagoIO. A widget can't manipulate data that are not within the Custom Widget settings.

vitorfdl commented on README.md:
```diff
@@ -1,29 +1,369 @@
+- **`TagoIO.deleteData(data, callback)`** - Delete device data
+- **`TagoIO.editResourceData(data, callback)`** - Edit platform resources
+
+#### Real-time Data
+- **`TagoIO.onRealtime(callback)`** - Receive real-time data updates
+- **`TagoIO.onSyncUserInformation(callback)`** - Access user context and authentication
+- **`TagoIO.onSyncBlueprintDevices(callback)`** - Access blueprint device configurations
```
The explanation is too short for the impact of the Realtime events. You must state those are async events that triggers based on Dashboard realtime events. For instance, `onRealtime` its called when new data is received through realtime events.

vitorfdl commented on README.md:
```diff
@@ -1,29 +1,369 @@
+
+**📁 [View All Examples](./examples/README.md)**
+
+### External Project Examples
+
+Complete project implementations:
+
+- **[Boilerplate Project](https://github.com/tago-io/custom-widget-boilerplate)**: Basic boilerplate project using Preact, showing fundamental SDK usage patterns.
+
+- **[SendData Widget](https://github.com/tago-io/custom-widget-example-send-data)**: Simple example demonstrating how to send data from your Custom Widget to TagoIO.
+
+- **[Wizard Widget](https://github.com/tago-io/custom-widget-example-wizard)**: Advanced demo showing a multi-step wizard built using TagoIO's Custom Widget SDK.
```
Don't we have a React custom widget to demonstrate React implementation ?

Freddyminu replied:
Couldn't find one using React, I've added the Echarts one to the example list.

vitorfdl commented on README.md:
````diff
@@ -1,29 +1,369 @@
+});
+```
+
+#### Data Validation
+Validate data before sending:
+```javascript
+function validateAndSendData(variable, value) {
+    if (!variable || value === undefined) {
+        throw new Error('Variable and value are required');
+    }
+    
+    if (typeof value === 'number' && isNaN(value)) {
+        throw new Error('Invalid numeric value');
+    }
+    
+    return window.TagoIO.sendData({ variable, value });
+}
+```
````
Not necessary at all.

vitorfdl commented on README.md:
````diff
@@ -1,29 +1,369 @@
+```
+
+#### Performance Optimization
+- Use `window.TagoIO.autoFill = true` to automatically handle device/bucket IDs
````
I have no idea what autoFill i doing. What handling device/bucket IDs means ?

gustavo-aguiar replied:
window.TagoIO.autoFill is a function used to automatically fill values in input to devices/bucket, you can manually send to another device or set true to send to the same origin, normally without this setting the default is true

vitorfdl commented on README.md:
````diff
@@ -1,29 +1,369 @@
+});
+```
+
+## 🎨 UI Design Guidelines
+
+### Responsive Design
+All examples include responsive CSS Grid layouts:
+```css
+.widget-container {
+    display: grid;
+    grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
+    gap: 20px;
+    padding: 20px;
+}
+```
+
+### TagoIO Design Language
+Use colors that complement TagoIO's interface:
+- Primary: `#2196F3`
+- Success: `#4CAF50`
+- Warning: `#FF9800`
+- Error: `#f44336`
+- Neutral: `#607D8B`
+
+### Accessibility
+- Use semantic HTML elements
+- Provide proper labels for form controls
+- Ensure sufficient color contrast
+- Support keyboard navigation
````
Not necessary at all.

vitorfdl commented on README.md:
````diff
@@ -1,29 +1,369 @@
+});
+```
+
+### Custom Header Configuration
+```javascript
+window.TagoIO.ready({
+    header: {
+        absolute: true,  // Position header absolutely
+        color: '#2196F3' // Custom header color
+    }
+});
+```
````
Never saw this pattern before. I would like to discuss this usage with @gustavo-aguiar

gustavo-aguiar replied:
I didn't see this styling inside the javascript, good catch @vitorfdl, the styling should be inline or in the CSS files

vitorfdl commented on README.md:
````diff
@@ -1,29 +1,369 @@
+npm run coverage
+```
+
+### Build Output
+
+The build process generates:
+- `dist/custom-widget.js` - Unminified library
+- `dist/custom-widget.min.js` - Minified library
+- `dist/custom-widget.css` - Optional styles
+- `dist/custom-widget.min.css` - Minified styles
````
Not necessary at all

vitorfdl commented on README.md:
```diff
@@ -1,29 +1,369 @@
+- **🐛 [Issue Tracker](https://github.com/tago-io/custom-widget/issues)** - Report bugs and request features
+- **📧 [Support Email](mailto:support@tago.io)** - Direct technical support
+
+### Useful Links
+
+- **[TagoIO Platform](https://admin.tago.io)** - Main platform dashboard
+- **[TagoIO Documentation](https://docs.tago.io)** - Complete platform documentation
+- **[API Reference](https://api.tago.io)** - REST API documentation
+- **[TagoRUN](https://run.tago.io)** - End-user portal documentation
```
Not necessary

vitorfdl commented on README.md:
```diff
@@ -1,29 +1,369 @@
+
+This project is licensed under the Apache-2.0 License - see the [LICENSE.md](LICENSE.md) file for details.
+
+## 🏷️ Version Information
+
+- **Current Version**: 1.1.0
+- **Node.js Compatibility**: >= 14.0.0
+- **Browser Support**: Modern browsers (ES6+)
+- **TypeScript**: Full type definitions included
```
Not necessary

vitorfdl commented on examples/README.md:
This file is unecessary and will only cause to have another dependancy to update. File names are already very descriptive.

Freddyminu committed:
PR Fixes
